src/author/dto/create.ts

```diff
@@ -0,0 +1,33 @@
+import { IsEmail, MaxLength, MinLength, IsNotEmpty } from "class-validator";
+
+export class CreateAuthorDto {
+  @MinLength(2)
+  @IsNotEmpty()
+  readonly name: string;
```

Comment:
Aqui é por conta da minha falta de conhecimento no typescript. Mas se a propriedade é readonly, como que o nest popula as informações?

Reply:
Como eu não to usando essa classe para criar instância e apenas para declarar o tipo (e ajudar nos auto-completes caso necessário) tudo bem ser readonly

Reply:
mas ele faz o binding dos parâmetros da requisição para os atributos do dto, não faz? Digo, sua requisição do postman tem um param que chama **name**, outro que chama **description** e mais um que chama **email**